File: 1764-form-array-by-concatenating-subarrays-of-another-array/1764-form-array-by-concatenating-subarrays-of-another-array.cpp
class Solution {
public:
    bool canChoose(vector<vector<int>>& groups, vector<int>& nums) {
         int idx=0;
        for(int i=0;i<nums.size()&&idx<groups.size();i++) {
            if(i+groups[idx].size()-1<nums.size()) {
                int j=0;
                for(j=0;j<groups[idx].size();j++) {
                    if(groups[idx][j]!=nums[i+j]) break;
                }
                if(j==groups[idx].size()) {
                    i+=groups[idx].size()-1;
                    idx++;
                }
            }
        }
        return idx>=groups.size();
    }
};

File: 1764-form-array-by-concatenating-subarrays-of-another-array/1764-form-array-by-concatenating-subarrays-of-another-array.java
class Solution {
    public boolean canChoose(int[][] groups, int[] nums) {
         int idx=0;
        for(int i=0;i<nums.length&&idx<groups.length;i++) {
            if(i+groups[idx].length-1<nums.length) {
                int j=0;
                for(j=0;j<groups[idx].length;j++) {
                    if(groups[idx][j]!=nums[i+j]) break;
                }
                if(j==groups[idx].length) {
                    i+=groups[idx].length-1;
                    idx++;
                }
            }
        }
        return idx>=groups.length;
    }
}